Testing theme and location reducers

Lvlad27 opened the pull request.

LvladCSol committed:
Initial commit

LvladCSol committed:
Init

LvladCSol committed:
change prettier indentation

LvladCSol committed:
update gitignore

LvladCSol committed:
add backend for api token data request

LvladCSol committed:
feat: serve token from api

LvladCSol committed:
add expiration time

LvladCSol committed:
adsa

LvladCSol committed:
Add token expiration time

LvladCSol committed:
update .gitignore

LvladCSol committed:
add geolocation function

LvladCSol committed:
testing the api

LvladCSol committed:
fix observations setResourceType call

LvladCSol committed:
add cors and dotenv packages

LvladCSol committed:
removed comment

LvladCSol committed:
chore: changed indentation

LvladCSol committed:
docs: change server script path

LvladCSol committed:
update geolocation function

LvladCSol committed:
add Inter fonts

LvladCSol committed:
add router and pages. Home Page main functionality

LvladCSol committed:
add initial css structure

LvladCSol committed:
add endpoints for api requests

LvladCSol committed:
remove comments

LvladCSol committed:
formatting

LvladCSol committed:
formatting

LvladCSol committed:
add some files

LvladCSol committed:
reinstall some packages

LvladCSol committed:
add extra variables for color & font size

LvladCSol committed:
moved to Home dir

LvladCSol committed:
add background color

LvladCSol committed:
add weather symbol icon endpoint

LvladCSol committed:
add axios config option to stringify response

LvladCSol committed:
adjust style for navbar component

LvladCSol committed:
add CurrWeatherCard comp. style & functionality

Lvlad27 committed:
Merge branch 'Leontescu_Vlad' into Vlad_Leontescu

LvladCSol committed:
comments cleaned

LvladCSol committed:
refactor: Loading and Error state

LvladCSol committed:
Merge branch 'Vlad_Leontescu' of github.com:Lvlad27/WeatherApp into HomePage

LvladCSol committed:
configure prettier indentation for 2 tabs

LvladCSol committed:
auto config indentation to 2 tabs

LvladCSol committed:
add JSON parsing to axios config

LvladCSol committed:
refactor

LvladCSol committed:
refactor: destructure props

LvladCSol committed:
minor change

LvladCSol committed:
docs: rename files, add symbol endpoint and function

LvladCSol committed:
add search bar component markup

LvladCSol committed:
add DailyWeather component

LvladCSol committed:
add weather details to main card component

LvladCSol committed:
change some styling

LvladCSol committed:
update for DailyWeather and Search children components

LvladCSol committed:
Merge remote-tracking branch 'upstream/Leontescu_Vlad' into Vlad_Leontescu

# Conflicts:
#	src/Components/CurrWeatherCard/CurrWeatherCard.jsx
#	src/Components/CurrWeatherCard/CurrWeatherCard.module.scss
#	src/Components/Navbar/Navbar.module.scss
#	src/Components/index.js
#	src/Helpers/endpoints.js
#	src/Helpers/geolocation.js
#	src/Pages/Home/Home.jsx
#	src/Pages/Home/Home.module.scss

LvladCSol committed:
refactor: formatDate is now a helper function

LvladCSol committed:
refactor: change props to currWeather.current

LvladCSol committed:
formatting update

LvladCSol committed:
search bar component

LvladCSol committed:
Merge branch 'Vlad_Leontescu' of github.com:Lvlad27/WeatherApp into Vlad_Leontescu

LvladCSol committed:
refactor: date formatting options

LvladCSol committed:
formatting

LvladCSol committed:
add search by location name endpoint

LvladCSol committed:
search bar autocomplete

LvladCSol committed:
Merge branch 'SearchBar' into Vlad_Leontescu

LvladCSol committed:
docs: folder and file restructure

LvladCSol committed:
add footer

LvladCSol committed:
change style and add main element

LvladCSol committed:
add hourly forecast component

LvladCSol committed:
current weather now changes based on search input

LvladCSol committed:
change header color

LvladCSol committed:
remove header styles comments

LvladCSol committed:
remove console.log

LvladCSol committed:
changes to webpack

LvladCSol committed:
restructure dataService files

LvladCSol committed:
Add Feedback Form component

Lvlad27 committed:
Merge branch 'Leontescu_Vlad' into FeedbackForm

LvladCSol committed:
Merge branch 'master' of github.com:Lvlad27/WeatherApp

LvladCSol committed:
fetched upstream

LvladCSol committed:
remove isSubmit state
move cross icon to Error Message Components
replace br tag with css

LvladCSol committed:
Merge branch 'FeedbackForm'

LvladCSol committed:
install react-icons

LvladCSol committed:
Add Favorite Cities Component

LvladCSol committed:
minor styling change

LvladCSol committed:
fix loadState and saveState

LvladCSol committed:
update FavPage

LvladCSol committed:
Merge branch 'FavoriteCities'

LvladCSol committed:
add themeContext

Lvlad27 committed:
Merge branch 'Leontescu_Vlad' into ThemeSwitch

Lvlad27 committed:
Update ThemeSwitch.jsx

LvladCSol committed:
Merge branch 'Leontescu_Vlad' of https://github.com/ximet/ReactTask

LvladCSol committed:
line format

LvladCSol committed:
Merge branch 'Leontescu_Vlad' of https://github.com/ximet/ReactTask

LvladCSol committed:
Merge branch 'master' into ThemeSwitch

LvladCSol committed:
Merge branch 'ThemeSwitch' of github.com:Lvlad27/WeatherApp into ThemeSwitch

LvladCSol committed:
resolved conflict

LvladCSol committed:
Dark/Light theme styling and scss refactoring.

LvladCSol committed:
Merge branch 'Leontescu_Vlad' of https://github.com/ximet/ReactTask into AboutPage

LvladCSol committed:
create About Page

LvladCSol committed:
add more images

LvladCSol committed:
More improvements

Add reusable Button component and refactor components accordingly.
Minor refactoring.

LvladCSol committed:
refactoring

remove save to local storage function
change ErrorMessage folder
change name of saved locations

LvladCSol committed:
refactor Redux for Theme and FavLocations

LvladCSol committed:
Combined reducers refactoring

- Theme and Favourite Cities states are now implemented through Redux and persisted in Local Storage
- Made some minor styling changes
- Remove unnecessary .env options

LvladCSol committed:
remove unnecessary imports

Lvlad27 committed:
Merge branch 'Leontescu_Vlad' into Refactor

LvladCSol committed:
Merge branch 'Leontescu_Vlad' of https://github.com/ximet/ReactTask into Refactor

LvladCSol committed:
remove themeContext

LvladCSol committed:
Merge branch 'Refactor' of github.com:Lvlad27/WeatherApp into Refactor

LvladCSol committed:
reformat

LvladCSol committed:
PR COMMENTS + custom hook

- remove console.log and themeContext from all Components
- add custom hook for all async requests

LvladCSol committed:
refactor themeSwitch input

LvladCSol committed:
add missing dispatch assignment line

LvladCSol committed:
refactor useApi for concurrent requests

LvladCSol committed:
refactor

- logo is now clickable and goes to homepage
- Message box now added for error and loading states
- set loading state when catching error

LvladCSol committed:
Merge branch 'Leontescu_Vlad' of https://github.com/ximet/ReactTask into Refactor

LvladCSol committed:
Merge branch 'master' into Refactor

LvladCSol committed:
test suites for theme and location reducers

JuliaShavyrova commented on src/__tests__/locations.test.js:
```diff
@@ -0,0 +1,10 @@
+    const action = { type: '' };
+    const result = cityReducer(favCityList, action);
+    expect(result).toEqual([]);
+  });
```
looks good, please add two more test cases (with different actions provided)

JuliaShavyrova commented on package.json:
```diff
@@ -83,6 +93,7 @@
     "pretty-quick": "1.11.1",
     "react-hot-loader": "4.13.0",
     "redux-immutable-state-invariant": "2.1.0",
+    "regenerator-runtime": "^0.13.9",
```
why do you need this package?

Lvlad27 replied:
I was getting similar error from this post and just followed the answer to try and fix the error:
[https://stackoverflow.com/questions/69091261/syntaxerror-support-for-the-experimental-syntax-decorators-legacy-isnt-curre](url). But it didn't work so I will remove it.

LvladCSol committed:
add test and edit package.json

- add 2 more test cases (diff. actions provided)
- remove  "regenerator-runtime": "^0.13.9"

JuliaShavyrova commented on src/__tests__/locations.test.js:
```diff
@@ -0,0 +1,25 @@
+  it('should delete city data object', () => {
+    const favCityList = mockCityData;
+    const cityName = 'Bucharest';
+    const action = { type: 'DELETE_FAV_CITY', payload: cityName };
```
did this test pass?
as far as i understand, your reducer removes location comparing ids, not city names

Lvlad27 replied:
Yes that's correct, but action.payload.locationData.id is the id of the given city name, action.payload being the city name. It compares each of the array's items id with the id of the given city name.

![image](https://user-images.githubusercontent.com/86689859/182649513-651b65fb-db38-4c9e-a5e1-208415dcfda0.png)
![image](https://user-images.githubusercontent.com/86689859/182648110-a7087508-cc48-4d83-81ed-e71697fe5990.png)

Lvlad27 replied:
fix test case and added extra one

![image](https://user-images.githubusercontent.com/86689859/182840521-bebf1067-bc77-4028-af6b-eaa1a4b10a3c.png)

LvladCSol committed:
fix test case #4 and add test case #5
